src/modules/Embed/Embed.js

```diff
@@ -0,0 +1,183 @@
+      customPropTypes.disallow(['sourceUrl']),
+      PropTypes.oneOf(_meta.props.source),
+    ]),
+
+    /** Specifies an id for source. */
+    sourceId: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.string,
+    ]),
+
+    /** Settings to configure video behavior. */
+    sourceSettings: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.shape({
+        /** Setting to true or false will force autoplay. */
+        autoPlay: PropTypes.bool,
+
+        /** Whether to show networks branded UI like title cards, or after video calls to action. */
+        brandedUI: PropTypes.bool,
+
+        /** Specifies a default chrome color with Vimeo or YouTube. */
+        color: PropTypes.string,
+
+        /** Whether to show networks branded UI like title cards, or after video calls to action. */
+        hd: PropTypes.bool,
+      }),
+    ]),
+
+    /** Specifies a url to use for embed. */
+    sourceUrl: customPropTypes.every([
+      customPropTypes.disallow(['source']),
+      PropTypes.string,
+    ]),
+  }
+
+  state = {}
+
+  handleClick = (e) => {
+    const { onClick } = this.props
+    const { active } = this.state
+
+    if (onClick) onClick(e, this.props)
+    if (!active) this.trySetState({ active: true })
+  }
+
+  getSource() {
+    const {
+      source,
+      sourceId,
+      sourceSettings = {},
+      sourceUrl,
+    } = this.props
+    const {
+      autoPlay = true,
+      brandedUI = false,
+      color = '#444444',
+      hd = true,
+    } = sourceSettings
+
+    if (source === 'youtube') {
+      return [
+        `//www.youtube.com/embed/${sourceId}`,
+        '?autohide=true',
+        `&amp;autoplay=${autoPlay}`,
+        `&amp;color=${encodeURIComponent(color)}`,
+        `&amp;hq=${hd}`,
+        '&amp;jsapi=false',
+        `&amp;modestbranding=${brandedUI}`,
+      ].join('')
+    }
+
+    if (source === 'vimeo') {
+      return [
+        `//player.vimeo.com/video/${sourceId}`,
+        '?api=false',
+        `&amp;autoplay=${autoPlay}`,
+        '&amp;byline=false',
+        `&amp;color=${encodeURIComponent(color)}`,
+        '&amp;portrait=false',
+        '&amp;title=false',
+      ].join('')
+    }
+
+    return sourceUrl
+  }
+
+  render() {
+    const {
+      aspectRatio,
+      children,
+      className,
+      icon,
+      placeholder,
+    } = this.props
+    const { active } = this.state
+
+    const classes = cx(
+      'ui',
+      aspectRatio,
+      useKeyOnly(active, 'active'),
+      'embed',
+      className,
+    )
+    const rest = getUnhandledProps(Embed, this.props)
+    const ElementType = getElementType(Embed, this.props)
+
+    return (
+      <ElementType {...rest} className={classes} onClick={this.handleClick}>
+        {Icon.create(icon)}
+        {placeholder && <img className='placeholder' src={placeholder} />}
+
+        <EmbedEmbed active={active} src={this.getSource()}>{children}</EmbedEmbed>
```

Comment:
Given that `EmbedEmbed`:

1. used in only one place, in `Embed`
1. only uses props from `Embed`
1. is not an exposed sub component

I would suggest using a method instead of a new component:

```jsx
renderEmbed = () => {
  const { active, children } = this.props
  if (!active) return null
  if (children) return <div className='embed'>{children}</div>

  return (
    <div className='embed'>
      <iframe
        allowFullScreen=''
        frameBorder='0'
        height='100%'
        scrolling='no'
        src={this.getSource()}
        width='100%'
      />
    </div>
  )
}

render() {
  return (
    // ...snip
    {this.renderEmbed}
  )
}
```